### PR TITLE
Add timers in routines that depend on cell-related work

### PR DESCRIPTION
**Pull request by mrowan137**
This PR adds CPU-side timers in routines that depend on number of cells.  This improves the measurement of costs for the CPU-side `algo.load_balance_costs_update = timers` option for updating load balance costs.  This fixes the previous behavior of assigning zero cost to cells with no particles when using `algo.load_balance_costs_update = timers`.

### CODE REVIEW

**Comment by mrowan137** [comment]
@RemiLehe @ax3l Are there any other routines that should be instrumented?

File: Source/Particles/Collision/PairWiseCoulombCollision.cpp [hunk 1]
```diff
@@ -50,13 +49,28 @@ PairWiseCoulombCollision::doCollisions (amrex::Real cur_time, MultiParticleConta
...
 #ifdef AMREX_USE_OMP
         info.SetDynamic(true);
 #pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
 #endif
         for (amrex::MFIter mfi = species1.MakeMFIter(lev, info); mfi.isValid(); ++mfi){
+            if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+            {
+                amrex::Gpu::synchronize();
+            }
+        amrex::Real wt = amrex::second();
```
Comment (ax3l) [inline Source/Particles/Collision/PairWiseCoulombCollision.cpp:64]:
```suggestion
            amrex::Real wt = amrex::second();
```

Comment (ax3l) [inline Source/Particles/Collision/PairWiseCoulombCollision.cpp]:
```suggestion
            amrex::Real wt = amrex::second();
```

**Comment by ax3l** [comment]
Awesome, thank you!

> Are there any other routines that should be instrumented?

Maybe some of the ionization physics here:
  https://github.com/ECP-WarpX/WarpX/blob/21.03/Source/Particles/MultiParticleContainer.cpp#L712-L725

And the QED physics here (cc @lucafedeli88):
  https://github.com/ECP-WarpX/WarpX/blob/21.03/Source/Particles/MultiParticleContainer.cpp#L1287-L1308
  https://github.com/ECP-WarpX/WarpX/blob/21.03/Source/Particles/MultiParticleContainer.cpp#L1348-L1373

**Comment by mrowan137** [comment]
Thanks, I added more instrumentation for those ionization and QED routines.

**Review by ax3l — APPROVED** [review]
Thank you! :rocket:

**Pull request merged**